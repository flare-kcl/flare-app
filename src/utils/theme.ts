import { createTheme } from '@shopify/restyle'

const palette = {
  purpleLight: '#8C6FF7',
  purplePrimary: '#5A31F4',
  purpleDark: '#3F22AB',

  greenLight: '#56DCBA',
  greenPrimary: '#0ECD9D',
  greenDark: '#0A906E',

  black: '#0B0B0B',
  white: '#F0F2F3',
}

const theme = createTheme({
  colors: {
    ...palette,
    mainBackground: palette.white,
    cardPrimaryBackground: palette.purplePrimary,
  },
  spacing: {
    s: 8,
    m: 16,
    l: 24,
    xl: 40,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  textVariants: {
    heading: {
      fontWeight: 'bold',
      fontSize: 30,
      lineHeight: 42.5,
      color: 'purplePrimary',
      alignSelf: 'center',
    },
  },
})

export type Theme = typeof theme
export default theme
