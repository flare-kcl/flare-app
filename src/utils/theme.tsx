import { ThemeProvider, createTheme } from '@shopify/restyle'

const palette = {
  purpleLight: '#8C6FF7',
  purplePrimary: '#5A31F4',
  purpleDark: '#3F22AB',

  greenLight: '#56DCBA',
  greenPrimary: '#0ECD9D',
  greenDark: '#0A906E',

  black: '#0B0B0B',
  white: '#F0F2F3',
}

const theme = createTheme({
  colors: {
    ...palette,
    mainBackground: palette.white,
    cardPrimaryBackground: palette.purplePrimary,
  },
  spacing: {
    0: 0,
    1: 4,
    2: 8,
    3: 12,
    4: 16,
    5: 20,
    6: 24,
    8: 32,
    10: 40,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  textVariants: {
    heading: {
      fontWeight: 'bold',
      fontSize: 30,
      lineHeight: 42.5,
      color: 'purplePrimary',
      alignSelf: 'center',
    },
  },
})

// Utility HOC for use in App and Tests
export const FlareThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export type Theme = typeof theme
export default theme
