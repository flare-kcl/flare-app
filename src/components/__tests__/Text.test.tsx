import { render } from '@testing-library/react-native'
import { FlareThemeProvider } from '@utils/theme'
import { Text } from '@components'

test('Text Renders Correctly', () => {
  const tree = render(
    <FlareThemeProvider>
      <Text />
    </FlareThemeProvider>,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
