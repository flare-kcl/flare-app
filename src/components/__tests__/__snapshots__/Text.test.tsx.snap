// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Renders Correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "display": "flex",
      },
    ]
  }
/>
`;
